### ISSUE TITLE
Feature Request: Handle TypeScript Better

### ISSUE THREAD

**Issue by jacob-israel-turner** [GoogleChromeLabs/ndb#34]
### Steps to reproduce

**Tell us about your environment:**

* ndb version: v1.0.6
* Platform / OS version: Ubuntu 16.04
* Node.js version: v8.10.0

**What steps will reproduce the problem?**

_Please include code that reproduces the issue._

I followed [this guide](https://hackernoon.com/debugging-javascript-typescript-node-apps-with-chrome-devtools-vs-code-and-webstorm-97b882aee0ad), hoping it would work out essentially the same.  It _kind of _ works.  I opened ndb with `ndb .` in the root of my project, then ran one of my npm scripts from the `ndb` UI.  The script was `ts-node src/index.ts` (I had to move my typings around a bit before this would work), and it ran just fine from inside `ndb`.  Unfortunately, after starting the script, I could not set breakpoints in any of the files listed by `ndb`.  However, if I stopped the script, set a breakpoint in one of the `.ts` files (specifically, `src/inxex.ts`), then restarted the script, ndb would start up and open a new file. This file matches what the article linked above led me to expect - the tab says the file is `index.ts?[sm]`.  However, I cannot actually _select_ this or other files from the list of files presented in the UI by ndb.

At this point, with the already-opened `index.ts?[sm]` file, I can move the breakpoint around and it appears to work fine.

**What is the expected result?**
If sourcemaps are available, the sourcemapped files should be available instead (or at least in addition to) the normal `.ts` files.

**What happens instead?**
They are not listed in the UI.

**Comment by jacob-israel-turner** [comment]
Also note, the sourcemapped files appear in the file finder (ctrl-p) as well.

**Comment by jacob-israel-turner** [comment]
It also appears that local changes are unable to be written to disk in `.ts` files.

![screenshot from 2018-07-23 15-58-00](https://www140.lunapic.com/do-not-link-here-use-hosting-instead/1532383372783645?5984605560)

**Comment by tangfang23** [comment]
Ok

**Comment by hinell** [comment]
Looks like chromium doesn't recognizes source maps currently. 
The Chrome 68 does fine for me as well (Win 10 x64).

**Comment by ak239** [comment]
Latest ndb version (1.0.17) contains a fix for `ts-node`. Please check it out and feel free to reopen this issue if it still a problem.

**Comment by hinell** [comment]
@ak239 Still doesn't work. If I use something like ``ndb node -r ts-node/register index.ts`` 
I get the following error after exiting program.:
<details>
 <summary>Expand errors</summary>
<pre>fs.js:119
    throw err;
    ^
Error: EPERM: operation not permitted, unlink 'UserName\AppData\Local\Temp\ndb-SU1uJo\BrowserMetrics\BrowserMetrics-5B60EFFF-2214.pma'
    at Object.unlinkSync (fs.js:925:3)
    at fixWinEPERMSync (UserName\AppData\Roaming\npm\node_modules\ndb\node_modules\rimraf\rimraf.js:211:13)
    at rimrafSync (UserName\AppData\Roaming\npm\node_modules\ndb\node_modules\rimraf\rimraf.js:311:28)
    at UserName\AppData\Roaming\npm\node_modules\ndb\node_modules\rimraf\rimraf.js:342:5
    at Array.forEach (<anonymous>)
    at rmkidsSync (UserName\AppData\Roaming\npm\node_modules\ndb\node_modules\rimraf\rimraf.js:341:26)
    at rmdirSync (UserName\AppData\Roaming\npm\node_modules\ndb\node_modules\rimraf\rimraf.js:334:7)
    at fixWinEPERMSync (UserName\AppData\Roaming\npm\node_modules\ndb\node_modules\rimraf\rimraf.js:209:5)
    at rimrafSync (UserName\AppData\Roaming\npm\node_modules\ndb\node_modules\rimraf\rimraf.js:311:28)
    at UserName\AppData\Roaming\npm\node_modules\ndb\node_modules\rimraf\rimraf.js:342:5
</pre>
</details>

**Comment by ak239** [comment]
Based on stack trace it looks like error at exit. Does source map work when ndb is running or is it immediately closed after start?

**Comment by jacob-israel-turner** [comment]
I just upgraded to v1.0.17. I do not get that error when exiting.  However, running `ts-node src/index.ts` or `node -r ts-node/register src/index.ts` do not make breakpoints work.  In fact, it's more broken than before - `index.ts?[sm]` no longer shows up in the fuzzy finder, so I cannot use the workaround anymore.

**Comment by ak239** [comment]
Does your ts file get little green dot next to its tab and if you try to set breakpoint inside this ts file, does breakpoint stay there are jump to some other file?
I will definitely debug it more.

**Comment by jacob-israel-turner** [comment]
Yes, all of my `.ts` files have a green dot on them.  The breakpoint stays in the `.ts` file I place it in, but it doesn't pause when executing.

**Comment by hinell** [comment]
@ak239 
The source maps are recognized but _breakpoints_ don't. The program doesn't pause at the breakpoint set in the ``index.ts`` file. When ``ndb`` is running there no errors is shown. Only after exiting program by the ``ctrl + c`` it.

**Comment by ak239** [comment]
Finally this one should be fixed in just published 1.0.20. Please check it out. 😄

**Comment by Yogu** [comment]
Sourcemaps are available for me, but breakpoints don't work (on Windows). It seems like a relative/absolute path handling error. If I click on a typescript line, a new file opens with the filename "<directory of file>/<absolute path of file>", e.g. "file:///C:/dir/C:\dir\file", which is empty.

Should I open a new issue? This can be reproduced with [cruddl](https://github.com/AEB-labs/cruddl) by running `ndb cruddl-dev.js`.

**Comment by ankur123** [comment]
I am seeing the same issue as well

**Comment by Aaron-Bird** [comment]
I also have this problem (1.1.5), is there any solution?
